WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerViewModel.kt

```diff
@@ -0,0 +1,72 @@
+package org.wordpress.android.ui.stories
+
+import android.content.Intent
+import androidx.lifecycle.ViewModel
+import org.wordpress.android.fluxc.model.LocalOrRemoteId.LocalId
+import org.wordpress.android.fluxc.model.PostImmutableModel
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.push.NotificationType
+import org.wordpress.android.ui.notifications.SystemNotificationsTracker
+import org.wordpress.android.ui.posts.EditPostRepository
+import org.wordpress.android.ui.posts.PostEditorAnalyticsSession
+import org.wordpress.android.util.analytics.AnalyticsUtilsWrapper
+import javax.inject.Inject
+
+class StoryComposerViewModel @Inject constructor(
+    private val systemNotificationsTracker: SystemNotificationsTracker,
+    private val saveInitialPostUseCase: SaveInitialPostUseCase,
+    private val analyticsUtilsWrapper: AnalyticsUtilsWrapper
+) :
+        ViewModel() {
+    private lateinit var editPostRepository: EditPostRepository
+    private lateinit var site: SiteModel
+    private lateinit var postEditorAnalyticsSession: PostEditorAnalyticsSession
+
+    fun start(
+        editPostRepository: EditPostRepository,
+        postId: LocalId,
+        intent: Intent,
+        postEditorAnalyticsSession: PostEditorAnalyticsSession?,
+        notificationType: NotificationType?
+    ) {
+        this.editPostRepository = editPostRepository
+
+        if (postId.value == 0) {
+            // Create a new post
+            saveInitialPostUseCase.saveInitialPost(editPostRepository, site)
+            // Bump post created analytics only once, first time the editor is opened
+            analyticsUtilsWrapper.trackEditorCreatedPost(
+                    intent.action,
+                    intent,
+                    site,
+                    editPostRepository.getPost()
+            )
+        } else {
+            editPostRepository.loadPostByLocalPostId(postId.value)
+        }
+
+        setupPostEditorAnalyticsSession(postEditorAnalyticsSession)
+
+        notificationType?.let {
+            systemNotificationsTracker.trackTappedNotification(it)
+        }
```

Comment:
I decided to put this in the `ViewModel` because it is business logic. We are saying that if the `notificationType` isn't null then we should track it since once it's not null it means we are resuming the `StoryComposerActivity` from a notification.

Reply:
👍